dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/CollectionRestRepository.java

```diff
@@ -139,4 +159,100 @@ public CollectionResource wrapResource(CollectionRest collection, String... rels
         return new CollectionResource(collection, utils, rels);
     }
 
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    protected CollectionRest createAndReturn(Context context) throws AuthorizeException {
+        HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
+        ObjectMapper mapper = new ObjectMapper();
+        CollectionRest collectionRest;
+        try {
+            ServletInputStream input = req.getInputStream();
+            collectionRest = mapper.readValue(input, CollectionRest.class);
+        } catch (IOException e1) {
+            throw new UnprocessableEntityException("Error parsing request body: " + e1.toString());
+        }
+
+        Collection collection;
+
+
+        try {
+            Community parent = null;
+            if (StringUtils.isNotBlank(req.getParameter("parentCommunity"))) {
+
+                UUID owningCommunityUuid = UUIDUtils.fromString(req.getParameter("parentCommunity"));
+                if (owningCommunityUuid == null) {
+                    throw new BadRequestException("The given owningCommunityUuid was invalid: "
+                            + req.getParameter("parentCommunity"));
+                }
+
+                parent = communityService.find(context, owningCommunityUuid);
+                if (parent == null) {
+                    throw new ResourceNotFoundException("Parent community for id: "
+                            + owningCommunityUuid + " not found");
+                }
+            }
+            collection = cs.create(context, parent);
+            cs.update(context, collection);
+            if (collectionRest.getMetadata() != null) {
+                for (MetadataEntryRest mer : collectionRest.getMetadata()) {
+                    String[] metadatakey = mer.getKey().split("\\.");
+                    cs.addMetadata(context, collection, metadatakey[0], metadatakey[1],
+                                   metadatakey.length == 3 ? metadatakey[2] : null, mer.getLanguage(), mer.getValue());
+                }
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
Please create a more useful RuntimeException message.  `e.getMessage()` can sometimes be "null", so this has the potential to return a strange null error.

Reply:
I would like to ask clarification on what we're expecting here. I've gone through most, if not all, repository classes and they've all included this behaviour when an SQLException has to be caught. Should we introduce an approach which is commonly accepted amongst the community as a solution to this issue? Seeing as this occurs in a lot of places which isn't just exclusive to this PR, should we fire off a new PR to solve all of these catch clauses all together in one bulk?

Reply:
The stack trace will show the value of SQLException.getMessage(), if any, so to use it as the message for the  RuntimeException is redundant and misses the opportunity to explain consequences of the failure which the lower-level code could not know.  What was this code unable to do *because* the DBMS did not do what was asked of it?

Reply:
Writing logic to properly specify and indicate what the actual issue was, would be too complex as an SQLException can be very broad. Temporarily I would suggest to leave this as-is and make a JIRA ticket to resolve all occurrences of these SQLException in one clean sweep.

Reply:
@Raf-atmire : What myself and @mwoodiupui are suggesting is actually a simple change.  I'd recommend simply changing this to be something like:

`throw new RuntimeException("Unable to create new Collection under parent Community " + parentCommunityString, e);`

This error message is much more informative to the user as it explains what this method was *trying to do* when the SQLException occurred, rather than just returning the database level error (with no additional information around how that error occurred).